### PR TITLE
Local Storage for Buckets

### PR DESCRIPTION
**Pull request by ayuishii**
Resolve TODO for `localStorage` alternative.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -1489,19 +1489,22 @@ const settingsBucket = await navigator.storageBuckets.openOrCreate("settings", {
...
-* RAM consumption limitation. Would need special quota system just for
-  LocalStorage. Buckets with small `quota` would get access then.
-* Non-issue: sync read API. We can read all data on `openOrCreate()`.
+This alternative was rejected because of `localStorage` does not follow
+the same durability policies, and we would need to create a quota system
+just to support it.
+
+Due to the synchronous nature of Web Storage API, user agents would need an
+in-renderer cache to support writes for `localStorage`.
+Therefore we would be able to support `"relaxed"` or `"strict"` policies.
```
Comment (pwnall) [inline explainer.md:1498]: we would -> we would not?

File: explainer.md [hunk 2]
```diff
@@ -1489,19 +1489,22 @@ const settingsBucket = await navigator.storageBuckets.openOrCreate("settings", {
...
+
+We would also need a special quota system just for `localStorage` and
```
Comment (pwnall) [inline explainer.md:1500]: "We" is a bit ambiguous here.

We could be the API designers, or browser vendors, or both. Either way, API designers (standards authors) and browser vendors are below web developers in the order of constituents. So, WDYT about strengthening the  argument by explaining that developers would need to reason about this extra quota system?

File: explainer.md [hunk 3]
```diff
@@ -1489,19 +1489,22 @@ const settingsBucket = await navigator.storageBuckets.openOrCreate("settings", {
...
+its RAM consumption. This could have performance implications of supporting multiple
+`localStorage` instances per origin. Each bucket could have a smaller quota than
+the existing quota limit but would add extra complexity to the implementation. 
+
+At first pass, the synchronous read API of `localStorage` seems like it could be
+a problem. However in this case, buckets could help from its main thread blocking
```
Comment (pwnall) [inline explainer.md:1506]: "buckets could help" feels like it has the wrong subject.

In this case, user agents don't need to block the main thread to read the `localStorage` contents. Instead, implementations can read the `localStorage` data asynchronously, while processing `navigator.storageBuckets.openOrCreate()`.

**Review by pwnall — CHANGES_REQUESTED** [review]
Thank you very much for this addition!

File: explainer.md (hunk 2 above)
Comment (ayuishii) [reply explainer.md]: I tried changing the perspective here to the web developer perspective.

**Comment by ayuishii** [comment]
Sorry for the delay, and thank you for the review!

**Comment by ayuishii** [comment]
Friendly ping...

File: explainer.md [hunk 4]
```diff
@@ -1489,20 +1489,25 @@ const settingsBucket = await navigator.storageBuckets.openOrCreate("settings", {
...
-
+This alternative was rejected because of `localStorage` does not follow
+the same durability policies, and we would need to create a quota system
+just to support it.
+
+Due to the synchronous nature of Web Storage API, user agents would need an
+in-renderer cache to support writes for `localStorage`.
+Therefore we would not be able to support `"relaxed"` or `"strict"` policies.
+
+Web developers would need to keeping track of separate quota just for `localStorage`
```
Comment (pwnall) [inline explainer.md:1500]: would need to keeping track -> would need to keep track

**Review by pwnall — COMMENTED** [review]
Thank you for tackling this difficult area! I hope the comments below are helpful!

File: explainer.md [hunk 5]
```diff
@@ -1489,20 +1489,25 @@ const settingsBucket = await navigator.storageBuckets.openOrCreate("settings", {
...
-TODO: Explain why we chose not to integrate `localStorage`.
-
-* LocalStorage doesn't follow durability policies. Sync write API means we need
-  in-renderer caches, so we couldn't support either `"relaxed"` or `"strict"`
-* RAM consumption limitation. Would need special quota system just for
-  LocalStorage. Buckets with small `quota` would get access then.
-* Non-issue: sync read API. We can read all data on `openOrCreate()`.
-
+This alternative was rejected because of `localStorage` does not follow
+the same durability policies, and we would need to create a quota system
```
[the remark this answers is not in the document]
Comment (ayuishii) [reply explainer.md]: I tried integrating your words into this section...

**Review by pwnall — APPROVED** [review]
LGTM.

Sorry, my view said this PR still has requested changes.

**Pull request merged**